Add HTTP server mode

emnoor-reef opened the pull request:
Starts a temporary HTTP server and waits for
request with seed to start job. After completing
the job, the server is shut down.

PS. Please ignore the commits. I'll squash them before merging :)

emnoor-reef committed:
Add HTTP server mode

Starts a temporary HTTP server and waits for
request with seed to start job. After completing
the job, the server is shut down.

# Conflicts:
#	src/compute_horde_prompt_solver/run.py

emnoor-reef committed:
Add HTTP server mode

Starts a temporary HTTP server and waits for
request with seed to start job. After completing
the job, the server is shut down.

emnoor-reef committed:
Add HTTP server mode

Starts a temporary HTTP server and waits for
request with seed to start job. After completing
the job, the server is shut down.

emnoor-reef committed:
Add HTTP server mode

Starts a temporary HTTP server and waits for
request with seed to start job. After completing
the job, the server is shut down.

emnoor-reef committed:
Add HTTP server mode

Starts a temporary HTTP server and waits for
request with seed to start job. After completing
the job, the server is shut down.

emnoor-reef committed:
Add HTTP server mode

Starts a temporary HTTP server and waits for
request with seed to start job. After completing
the job, the server is shut down.

emnoor-reef committed:
Add HTTP server mode

Starts a temporary HTTP server and waits for
request with seed to start job. After completing
the job, the server is shut down.

emnoor-reef committed:
Add HTTP server mode

Starts a temporary HTTP server and waits for
request with seed to start job. After completing
the job, the server is shut down.

emnoor-reef commented on src/compute_horde_prompt_solver/run.py:
```diff
@@ -45,6 +49,17 @@ def parse_arguments():
     parser.add_argument(
         "--seed", type=int, default=42, help="Random seed for reproducibility"
     )
+    parser.add_argument(
+        "--server",
+        action="store_true",
+        help="Spin up a temporary HTTP server to receive the seed",
+    )
```
Should seed and server be mutually exclusive? Currently passing both seed and server would make it use the server and ignore the seed.

andreea-popescu-reef replied:
yeah should throw at least a warning if both server and seed are given

to be nitpicky we should also warn if server_port is passed without server as it would do nothing

emnoor-reef commented on src/compute_horde_prompt_solver/run.py:
```diff
@@ -91,20 +106,78 @@ def process_file(
 
     args.output_dir.mkdir(parents=True, exist_ok=True)
 
+    # TODO: make sure if it is ok to load model before calling set_deterministic()!!
     model = setup_model(args.model)
```
I made 2-3 runs with same/different seeds and they seem to be behaving similarly. Sill I am a bit skeptical.

emnoor-reef committed:
Make --seed and --server mutually exclusive

Also remove --seed default value. So either --seed or --server must be specified.
